compiler/src/org.graalvm.compiler.core.aarch64/src/org/graalvm/compiler/core/aarch64/AArch64NodeMatchRules.java

```diff
@@ -25,28 +25,58 @@
 
 package org.graalvm.compiler.core.aarch64;
 
+import jdk.vm.ci.meta.AllocatableValue;
 import jdk.vm.ci.meta.JavaKind;
+import jdk.vm.ci.meta.Value;
 import org.graalvm.compiler.asm.aarch64.AArch64MacroAssembler;
+import org.graalvm.compiler.core.common.LIRKind;
 import org.graalvm.compiler.core.gen.NodeMatchRules;
 import org.graalvm.compiler.core.match.ComplexMatchResult;
 import org.graalvm.compiler.core.match.MatchRule;
+import org.graalvm.compiler.graph.Node;
 import org.graalvm.compiler.lir.LIRFrameState;
+import org.graalvm.compiler.lir.Variable;
 import org.graalvm.compiler.lir.aarch64.AArch64ArithmeticOp;
 import org.graalvm.compiler.lir.gen.LIRGeneratorTool;
 import org.graalvm.compiler.nodes.ConstantNode;
 import org.graalvm.compiler.nodes.DeoptimizingNode;
 import org.graalvm.compiler.nodes.NodeView;
 import org.graalvm.compiler.nodes.ValueNode;
 import org.graalvm.compiler.nodes.calc.AddNode;
+import org.graalvm.compiler.nodes.calc.AndNode;
 import org.graalvm.compiler.nodes.calc.BinaryNode;
 import org.graalvm.compiler.nodes.calc.LeftShiftNode;
+import org.graalvm.compiler.nodes.calc.NotNode;
+import org.graalvm.compiler.nodes.calc.OrNode;
 import org.graalvm.compiler.nodes.calc.RightShiftNode;
+import org.graalvm.compiler.nodes.calc.SubNode;
 import org.graalvm.compiler.nodes.calc.UnsignedRightShiftNode;
+import org.graalvm.compiler.nodes.calc.XorNode;
 import org.graalvm.compiler.nodes.memory.Access;
 
 import jdk.vm.ci.aarch64.AArch64Kind;
 
+import java.util.HashMap;
+import java.util.Map;
+
 public class AArch64NodeMatchRules extends NodeMatchRules {
+    private static final Map<Class<? extends Node>, AArch64ArithmeticOp> nodeOpMap;
+
+    private static final Map<Class<? extends BinaryNode>, AArch64MacroAssembler.ShiftType> shiftTypeMap;
+
+    static {
+        nodeOpMap = new HashMap<>();
```

Comment:
As a general rule of thumb, we want to avoid usages of `java.util.HashMap` in Graal. Instead we have our own Map implementation which is created with`org.graalvm.collections.EconomicMap.create(Equivalence.IDENTITY, 5)`. Since the element count is below 8, the EconomicMap implementation does a linear search over the entities, which is pretty similar to a `if/elseif/else` cascade.